api/question.go

```diff
@@ -48,28 +50,31 @@ func GetQuestionHandler(res http.ResponseWriter, req *http.Request) {
 		httputil.WriteError(res, http.StatusNotFound, "question not found")
 		return
 	}
+
 	var answers []models.Answer
-	if err := db.Raw(ANSWERS_QUERY, question.ID).Scan(&answers).Error; err != nil {
+
+	preloadingString := strings.Repeat("Replies.", RepliesDepth)
+
+	votes_subquery := db.Table("votes").
+		Select("votes.answer, COUNT(CASE votes.vote WHEN ? THEN 1 ELSE NULL END) as upvotes, COUNT(CASE votes.vote WHEN ? THEN 1 ELSE NULL END) as downvotes", VoteUp, VoteDown).
+		Group("votes.answer")
+
+	err = db.Table("answers").
+		Select("answers.*, vote_counts.upvotes, vote_counts.downvotes").
+		Where("answers.deleted_at IS NULL AND answers.parent IS NULL AND answers.question = ?", question.ID).
+		Joins("LEFT JOIN (?) vote_counts ON vote_counts.answer = answers.id", votes_subquery).
+		Preload(preloadingString[:len(preloadingString)-1], func(db *gorm.DB) *gorm.DB {
+			// perform join also on preloaded replies so they have their respective votes
+			return db.Select("answers.*, vote_counts.upvotes, vote_counts.downvotes").
+				Joins("LEFT JOIN (?) vote_counts ON vote_counts.answer = answers.id", votes_subquery)
+		}).
+		Find(&answers).Error
+
+	if err != nil {
```

Comment:
It's the exact same code found in `api/answer.go`. Can we factor it and avoid duplication?